src/components/QRCodeScanner/QRCodeScanner.jsx

```diff
@@ -0,0 +1,270 @@
+import React, { useState, useEffect, useRef } from 'react';
```

Comment:
When refactoring or adding code, I think we should use always use typescript

Reply:
I agree with you. We usually refactor with TS. We'll definitely look into that. Do you think it is suitable to make that change in this PR?

Reply:
not necessarily in this PR